packages/inform/src/Adaptor.js

```diff
@@ -0,0 +1,256 @@
+      options
+    );
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      `data/${resolvedFormId}`,
+      {
+        query: {
+          ...resolvedOptions,
+        },
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Get a single data submission for a single form
+ * @example
+ * getSubmission('6225', '7783155')
+ * @function
+ * @public
+ * @param {string} formId - Id of the form's submissions to be retrieved.
+ * @param {string} submissionId - Id of the submission to be retrieved.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function getSubmission(formId, submissionId) {
+  return async state => {
+    const [resolvedFormId, resolvedSubmissionId] = expandReferences(
+      state,
+      formId,
+      submissionId
+    );
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      `data/${resolvedFormId}/${resolvedSubmissionId}`
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Fetch a single attachment's metadata
+ * @example
+ * getAttachmentMetadata('621985')
+ * @function
+ * @public
+ * @param {string} attachmentId - Id of the attachment to be retrieved.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function getAttachmentMetadata(attachmentId) {
+  return async state => {
+    const [resolvedAttachmentId] = expandReferences(state, attachmentId);
+
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      `media/${resolvedAttachmentId}`
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * Download an attachment in binary or base64 format
+ * @example <caption>Download an attachment</caption>
+ * downloadAttachment('622038', {
+ *   filename:
+ *     'unicefbih/attachments/download_1.png',
+ * });
+ * @example <caption>Download an attachment in base64 format</caption>
+ * downloadAttachment('622038', {
+ *   filename:
+ *     'unicefbih/attachments/download_1.png',
+ *   parseAs: 'base64',
+ * });
+ * @function
+ * @public
+ * @param {string} attachmentId - Id of the attachment to be retrieved.
+ * @param {object} options - Optional request options. Supported options are: `filename` for the specific attachment to be downloaded, and `parseAs` for either 'stream` or 'base64`. Defaults to `parseAs: stream`.
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function downloadAttachment(attachmentId, options = {}) {
+  return async state => {
+    const [resolvedAttachmentId, resolvedOptions] = expandReferences(
+      state,
+      attachmentId,
+      options
+    );
+
+    let response = await util.request(
+      state.configuration,
+      'GET',
+      `files/${resolvedAttachmentId}`,
+      {
+        query: {
+          filename: resolvedOptions?.filename,
+        },
+      }
+    );
+
+    const { headers } = response;
+
+    // Default to stream if no parseAs option is provided
```

Comment:
Ah this is isn't what I meant.

We need to explain why we're adding manual redirect logic and not just passing `maxRedirections`.